app/assets/javascripts/application.js.erb

```diff
@@ -576,13 +551,67 @@ onPageLoad(() => {
     }
   });
 
+  on('click', 'button.comment-cancel', (event) => {
+    const comment = (parentSelector(event.target, '.comment'));
+    const commentId = comment.getAttribute('data-shortid');
+    if (commentId !== null && commentId !== '') {
+      fetch('/comments/' + commentId + '?show_tree_lines=true')
+        .then(response => {
+          response.text().then(text => replace(comment, text));
+        });
+    } else {
+      comment.remove();
+    }
+  });
+
   on('click', 'a.comment_editor', (event) => {
     let comment = parentSelector(event.target, '.comment');
-    fetch('/comments/' + comment.getAttribute('data-shortid') + '/edit')
+    const commentId = comment.getAttribute('data-shortid')
+    fetch('/comments/' + commentId + '/edit')
       .then(response => {
-        return response.text().then(function(text) {
-          replace(comment, text);
+        response.text().then(text => replace(comment, text));
+      });  
+    removeExtraInput(comment); 
+  });
+
+  on("click", "a.comment_deletor", (event) => {
+    event.preventDefault();
+    if (confirm("Are you sure you want to delete this comment?")) {
+      const comment = parentSelector(event.target, '.comment');
+      removeExtraInput(comment);
```

Comment:
I think you have to move this down after the fetched comment is inserted into the page. Same with the other calls to the cleanup function.

Reply:
I moved the `removeExtraInput()` into the replace function to deal with an async issue and to reduce the number of place it is called.  It works on all of the comment calls except for the post/edit comment. I haven't rewritten that one yet, to remove the jQuery.